lib/chainweb/json.ex

```diff
@@ -0,0 +1,53 @@
+defprotocol Kadena.Chainweb.JSON do
+  alias Kadena.Types.{
+    ListenRequestBody,
+    LocalRequestBody,
+    PollRequestBody,
+    SendRequestBody,
+    SPVRequestBody
+  }
+
+  @type json_string :: String.t()
+  @type request_body ::
+          ListenRequestBody.t()
+          | LocalRequestBody.t()
+          | PollRequestBody.t()
+          | SendRequestBody.t()
+          | SPVRequestBody.t()
+
+  @spec parse(request_body :: request_body()) :: json_string()
+  def parse(request_body)
+end
+
+alias Kadena.Chainweb.JSON
+
+defimpl JSON, for: Kadena.Types.LocalRequestBody do
```

Comment:
@santiagoboterokommit IMO, this protocol definition for structs may work better https://hexdocs.pm/elixir/1.14/Protocol.html#module-protocols-and-structs , what do you think?